src/connection_manager.rs

```diff
@@ -780,34 +764,41 @@ mod test {
          }
      }
 
-    fn make_temp_config(beacon_port: Option<u16>) -> (PathBuf, TempDir) {
+    struct TestConfigFile {
+        pub path: PathBuf
+    }
+
+    impl Drop for TestConfigFile {
+        fn drop(&mut self) {
+            let _ = remove_file(&self.path);
+        }
+    }
+
+    fn make_temp_config(beacon_port: Option<u16>) -> TestConfigFile {
```

Comment:
TestConfigFile and TempDir are both used as raii we can get rid of TempDir.
Address this as a part of cleanup